### PR TITLE
package name parameter is now optional

### PR DESCRIPTION
**Pull request by luisvsilva**
Added test to check start with package name and without.

starting with package name outputs a deprecation warning to the console.

### CODE REVIEW

File: www/pushlink.js [hunk 1]
```diff
@@ -44,19 +44,32 @@ var PushLink = function() {
...
  * @param {string} apiKey - Your PushLink API key
  * @param {string} deviceId - The device id
  * @param {function} successCallback - A function to be called if the command succeeded
  * @param {function} errorCallback - A function to be called if the command failed
  */
-PushLink.prototype.start = function(apiKey, deviceId, successCallback, errorCallback) {
+PushLink.prototype.start = function(packageName,apiKey, deviceId, successCallback, errorCallback) {
   'use strict';
 
+  if (!errorCallback){
```
Comment (rscarvalho) [inline www/pushlink.js:56]: What if `errorCallback` is `null`? It may be better to check for the length of `arguments` here.

Comment (rscarvalho) [inline www/pushlink.js:56]: I'd suggest the comparison should be something like this:

```js
if (arguments.length === 4 && typeof deviceId !== 'string') {
  // then the signature was (apiKey, deviceId, successCallback, errorCallback)
  apiKey = arguments[0];
  deviceId = arguments[1];
  successCallback = arguments[2];
  errorCallback = arguments[3];
}
```

Then we don't need the extra `cordoaExec` call.

**Comment by celiovasconcelos** [comment]
Hi Luis,

I think the Rodolfos's PR is better. https://github.com/pushlink/pushlink-phonegap/pull/18

I will wait for your comments on that PR.

About StartAction.java. There is no need test the argument there. I think it's safe to always get the package from application context. Right?

Thank you.

**Comment by luisvsilva** [comment]
Definitely better!!!

It is safe, it just feels a bit dirty to ignore the parameter :), but yeah
I think there is no reason why you would want to use a different package
name than the one you are currently in (is that even possible)?

Thanks Guys!!

Luis

**Pull request closed without merging**